receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/config.go

```diff
@@ -23,15 +25,93 @@ import (
 type Config struct {
 	internal.ConfigSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 
-	// Include specifies a filter on the devices that should be included from the generated metrics.
-	// Exclude specifies a filter on the devices that should be excluded from the generated metrics.
-	// If neither `include` or `exclude` are set, metrics will be generated for all devices.
-	Include MatchConfig `mapstructure:"include"`
-	Exclude MatchConfig `mapstructure:"exclude"`
+	// IncludeDevices specifies a filter on the devices, filesystem types or mount points that should be included from the generated metrics.
+	IncludeDevices DeviceMatchConfig `mapstructure:"include_devices"`
+	// ExcludeDevices specifies a filter on the devices, filesystem types or mount points that should be excluded from the generated metrics.
+	ExcludeDevices DeviceMatchConfig `mapstructure:"exclude_devices"`
+
+	// IncludeFSTypes specifies a filter on the filesystem types that should be included from the generated metrics.
+	IncludeFSTypes FSTypeMatchConfig `mapstructure:"include_fs_types"`
+	// ExcludeFSTypes specifies a filter on the filesystem types points that should be excluded from the generated metrics.
+	ExcludeFSTypes FSTypeMatchConfig `mapstructure:"exclude_fs_types"`
+
+	// IncludeMountPoints specifies a filter on the mount points that should be included from the generated metrics.
+	IncludeMountPoints MountPointMatchConfig `mapstructure:"include_mount_points"`
+	// ExcludeMountPoints specifies a filter on the mount points that should be excluded from the generated metrics.
+	ExcludeMountPoints MountPointMatchConfig `mapstructure:"exclude_mount_points"`
 }
 
-type MatchConfig struct {
+type DeviceMatchConfig struct {
 	filterset.Config `mapstructure:",squash"`
 
 	Devices []string `mapstructure:"devices"`
 }
+
+type FSTypeMatchConfig struct {
+	filterset.Config `mapstructure:",squash"`
+
+	FSTypes []string `mapstructure:"fs_types"`
+}
+
+type MountPointMatchConfig struct {
+	filterset.Config `mapstructure:",squash"`
+
+	MountPoints []string `mapstructure:"mount_points"`
+}
+
+func (cfg *Config) newFilter() (*fsFilter, error) {
+	var err error
+	filter := fsFilter{}
+
+	filter.includeDevice, err = newIncludeFilterHelper(cfg.IncludeDevices.Devices, &cfg.IncludeDevices.Config, "device")
```

Comment:
I'd update all the strings in this function to use the `consts` defined in metadata